Integrations/ThreatConnect/ThreatConnect.py

```diff
@@ -0,0 +1,1338 @@
+
+    return tc.api_request(ro).json()
+
+
+def tc_get_events():
+    raw_response = get_events_request()
+    data = raw_response.get('data', {}).get('event', {})
+    content = []
+    headers = ['ID', 'Name', 'OwnerName', 'EventDate', 'DateAdded', 'Status']
+
+    for event in data:
+        content.append({
+            'ID': event.get('id'),
+            'Name': event.get('name'),
+            'OwnerName': event.get('ownerName'),
+            'DateAdded': event.get('dateAdded'),
+            'EventDate': event.get('eventDate'),
+            'Status': event.get('status')
+        })
+    context = {
+        'TC.Event(val.ID && val.ID === obj.ID)': content
+    }
+
+    return_outputs(tableToMarkdown('ThreatConnect Events', content, headers, removeNull=True), context, raw_response)
+
+
+def tc_get_indicator_types_request():
+    tc = get_client()
+    ro = RequestObject()
+    ro.set_http_method('GET')
+    ro.set_request_uri('/v2/types/indicatorTypes')
+
+    return tc.api_request(ro).json()
+
+
+def tc_get_indicator_types():
+    raw_response = tc_get_indicator_types_request()
+    data = raw_response.get('data', {}).get('indicatorType', {})
+    content = []
+    headers = ['Name', 'Custom', 'Parsable', 'ApiBranch', 'CasePreference', 'value1Label', 'Value1Type']
+
+    for type in data:
+        content.append({
+            'Custom': type.get('custom'),
+            'Name': type.get('name'),
+            'Parsable': type.get('parsable'),
+            'ApiBranch': type.get('apiBranch'),
+            'ApiEntity': type.get('apiEntity'),
+            'CasePreference': type.get('casePreference'),
+            'value1Label': type.get('value1Label'),
+            'Value1Type': type.get('value1Type')
+        })
+    context = {
+        'TC.Indicator(val.Name && val.Name === obj.Name)': content
+    }
+
+    return_outputs(tableToMarkdown('ThreatConnect Indicator types', content, headers, removeNull=True), context,
+                   raw_response)
+
+
+def associate_indicator_request(indicator_type, indicator, group_type, group_id):
+    tc = get_client()
+    ro = RequestObject()
+    ro.set_http_method('POST')
+    ro.set_request_uri('/v2/indicators/{}/{}/groups/{}/{}'.format(indicator_type, indicator, group_type, group_id))
+    response = tc.api_request(ro).json()
+
+    return response
+
+
+def associate_indicator():
+    group_id = int(demisto.args().get('group_id'))
+    group_type = demisto.args().get('group_type')
+    indicator_type = demisto.args().get('indicator_type')
+    indicator = demisto.args().get('indicator')
+
+    response = associate_indicator_request(indicator_type, indicator, group_type, group_id)
+
+    if response.get('status') == 'Success':
+        contents = {
+            'IndicatorType': indicator_type,
+            'Indicator': indicator,
+            'GroupType': group_type,
+            'GroupID': group_id
+        }
+    else:
+        return_error(response.get('message'))
```

Comment:
need to make sure `response` got `message` key

Reply:
?

Reply:
Checked. It will return an error message.